### PR TITLE
[IDEAS] Proposal to allow using wood forests for sneak attacks.

### PR DESCRIPTION
**Pull request by vaibhavkaul**
Also removed a bunch of unnecessary periods

### CODE REVIEW

**Comment by vaibhavkaul** [comment]
@TheJJ this was always my biggest annoyance while sneaking a villager in black forest.

**Comment by TheJJ** [comment]
To have Kevin stop complaining, please [rebase](https://github.com/SFTtech/openage/blob/master/doc/contributing.md#rebasing) and fix the whitespace error it tells you about.

**Comment by vaibhavkaul** [comment]
@TheJJ Kevin is good now.

**Comment by TheJJ** [comment]
yay :) now for the rebase!

**Comment by vaibhavkaul** [comment]
Ok. Following the instructions. Should be rebased!

**Comment by elnabo** [comment]
I think there more bad than good from this idea.

- It is a micro intensive protection against raiding.
- It is a confusing mechanic for low micro players (or new to this concept). Why did my villagers disapears, how to get it back (assuming they notice).
- Raise few tricky questions, what happen when the tree is cut by an other villager (where does the villager go ?) or an onager (does he die ?) or what happen if an enemy villager want to go into an occupied one.
- It is an easy way to grief, hide a lot of villagers into a lot of trees.
- Increase save file sizes

**Comment by TheJJ** [comment]
as always, such a feature would be opt-in when somebody creates a game!

**Comment by elnabo** [comment]
That still leaves the question of tree cutting.

**Comment by vaibhavkaul** [comment]
@elnabo Thats a very fair concern. This is literally the first thing that came to my mind when I saw the ideas doc. Totally understand if this would be too complicated for in game mechanics. 

As a fan of drush / trush attacks, being able to sneak in villagers was just always something i really liked doing. 

Thanks!

**Comment by heinezen** [comment]
The motto of our ideas is "crazy XOR good", so this is a pretty fitting addition.

> It is a confusing mechanic for low micro players (or new to this concept). Why did my villagers disapears, how to get it back (assuming they notice).

It could have it's own game mode. Besides that, SWGB has units that can go invisible and this idea is not that far off from that.

> Raise few tricky questions, what happen when the tree is cut by an other villager (where does the villager go ?) or an onager (does he die ?) or what happen if an enemy villager want to go into an occupied one.

I guess the villager will be revealed. Makes sense that you would spot a fake tree when you cut it down.

This could be a a lot of fun in custom game modes.

**Comment by elnabo** [comment]
I believe some kind of Camouflage unit (behaving like a invisible transportship for a single villager) would be a simpler idea.

**Comment by TheJJ** [comment]
The "camouflage" ability should be relatively easy to implement, which we can extend to certain conditions then, like being near a large-enough forest. It could also be time-based with a cool-down period.

**Comment by vaibhavkaul** [comment]
I like the time based idea so both players are required to micro better in this game mode. Making sure the strategy does not become too OP

**Comment by vaibhavkaul** [comment]
@TheJJ should be all good now.

**Comment by TheJJ** [comment]
Almost! You somehow duplicated the commits, now there are four of them. You can do the rebase again and use the 'fixup' feature to meld those commits into one (`git rebase -i upstream/master`, then use the `fixup` command).

**Comment by ChipmunkV** [comment]
Do the trees in AoE block vision? If they do then you can design the map with small paths and pockets in the forests and hide there (like Dota2 juking). Btw, there is another mechanic from Dota2 with simple implementation: the Treant unit who becomes invisible near the trees.

**Comment by elnabo** [comment]
I don't think anything block vision in AoE2.

File: doc/ideas/gameplay.md [hunk 1]
```diff
@@ -18,37 +18,45 @@ Environment
...
  - Chemistry industry: Coffeine, Peniciline, ...?
 
+### Forests
+
+ - A single villager can hide (per some tile area) in the forest for sneak attacks. Maybe can only visible to a certain type of unit. But invisible to other villagers (until they pop out and start building stuff)
+
+### Forests
+
+ - A single villager can hide (per some tile area) in the forest for sneak attacks. Maybe can only visible to a certain type of unit. But invisible to other villagers (until they pop out and start building stuff)
+
```
Comment (TheJJ) [inline doc/ideas/gameplay.md]: Now this section is duplicated.

Comment (vaibhavkaul) [reply doc/ideas/gameplay.md]: Damn, really appreciate the patience. Let me try that one more time.

**Comment by vaibhavkaul** [comment]
@TheJJ again, thanks for your patience. Looks good now.

Love this project!

**Pull request merged**